### DISCUSSION TITLE
Help structuring data in directus

### DISCUSSION THREAD

**Discussion by cupcakearmy** [directus/directus#2843] [Archive]
So I have a simple task, but I can't wrap my head around it how to model it in directus

Think of recipes. recipes have ingredient.
Both `recipes` and `ingredients` are a collection.
Now I need to reference a variable amount of ingredients, each associated with a quantity string.

The repeater element does not allow for other collections yet (https://github.com/directus/app/issues/1996)
Any way to do this another way?

Thanks!

**Comment by rijkvanzanten** [comment]
The way you'd set it up in a database is by adding the quantity as a field on the junction table in a many to many. I'm not a 100% sure if v8's m2m edit modal shows those extra junction table fields though

**Reply by cupcakearmy** [reply]
Thanks for the quick answer :)
I'll try it tomorrow

**Reply by rijkvanzanten** [reply]
Just double checked it's a thing in v9 https://cln.sh/xjdtCa

**Reply by cupcakearmy** [reply]
Beautiful! I just tried in the latest v8 but that is not a thing there. Gonna wait for v9 then 👍

**Reply by cupcakearmy** [reply]
Quick question: The junction collections don't have an auto generate option yet in v9 right? (wanted to test myself) @rijkvanzanten

**Reply by rijkvanzanten** [reply]
No not yet, though it won't be long before it gets there. Just setting up this demo was already driving me crazy haha
